apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/evaluations/[evaluationId]/_components/EvaluationResults/EvaluationResultInfo/index.tsx

```diff
@@ -84,53 +78,40 @@ export function EvaluationResultInfo({
   evaluationResult: EvaluationResultWithMetadata
   providerLog?: ProviderLogDto
 }) {
-  const ref = useRef<HTMLDivElement>(null)
-  const [selectedTab, setSelectedTab] = useState<string>('metadata')
   const [selected, setSelected] = useState<MaybeDocumentLog>(null)
   const onClickOpen = useCallback(async () => {
     setSelected(evaluationResult.documentLogId)
   }, [evaluationResult.documentLogId])
-  const height = useDynamicHeight({ ref, paddingBottom: 16 })
   return (
     <>
-      <div
-        ref={ref}
-        className='relative w-80 flex-shrink-0 flex flex-col border border-border rounded-lg items-center custom-scrollbar overflow-y-auto'
-        style={{
-          maxHeight: height ? `${height}px` : 'auto',
-        }}
-      >
-        <div className='z-10 w-full sticky top-0 px-4 bg-white flex justify-center'>
-          <div className='pt-6'>
-            <TabSelector
-              options={[
-                { label: 'Metadata', value: 'metadata' },
-                { label: 'Messages', value: 'messages' },
-              ]}
-              selected={selectedTab}
-              onSelect={setSelectedTab}
-            />
+      <MetadataInfoTabs
+        className='w-80'
+        beforeTabs={
+          <div>
+            <Text.H6 color='foregroundMuted'>
+              Showing the evaluation metadata. Check the{' '}
+            </Text.H6>
+            <button onClick={onClickOpen}>
+              <Text.H6 color='primary'>original log metadata here</Text.H6>
+            </button>
           </div>
-        </div>
-        <div className='px-4 flex flex-col relative w-full max-w-full'>
-          {selectedTab === 'metadata' && (
-            <EvaluationResultMetadata
-              evaluation={evaluation}
-              evaluationResult={evaluationResult}
-              providerLog={providerLog}
-            />
-          )}
-          {selectedTab === 'messages' && (
-            <EvaluationResultMessages providerLog={providerLog} />
-          )}
-        </div>
-        <div className='w-full py-4 sticky bottom-0 bg-white flex justify-center'>
-          <Button variant='link' onClick={onClickOpen}>
-            Check original log
-            <Icon name='arrowRight' widthClass='w-4' heightClass='h-4' />
-          </Button>
-        </div>
-      </div>
+        }
+      >
+        {({ selectedTab }) => (
+          <>
+            {selectedTab === 'metadata' && (
+              <EvaluationResultMetadata
+                evaluation={evaluation}
+                evaluationResult={evaluationResult}
+                providerLog={providerLog}
+              />
+            )}
+            {selectedTab === 'messages' && (
+              <EvaluationResultMessages providerLog={providerLog} />
+            )}
+          </>
+        )}
```

Comment:
Why the function?

Reply:
This is a render prop. `selectedtab` comes from inside parent component